src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -374,13 +379,14 @@ public void fireFileStarted(String fileName) {
      * Notify all listeners about the errors in a file.
      *
      * @param fileName the audited file
-     * @param errors the audit errors from the file
+     * @param fileProcessingResult file processing results object
      */
     @Override
-    public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
+    public void fireErrors(String fileName, FileProcessingResult fileProcessingResult) {
         final String stripped = CommonUtils.relativizeAndNormalizePath(basedir, fileName);
         boolean hasNonFilteredViolations = false;
-        for (final LocalizedMessage element : errors) {
+        final SortedSet<LocalizedMessage> auditErrors = fileProcessingResult.getMessages();
+        for (LocalizedMessage element : auditErrors) {
```

Comment:
Rename element to errorMessage, errors to errorMessages.  We loop over messages, not audit events